lms/views/basic_lti_launch.py

```diff
@@ -166,6 +168,50 @@ def db_configured_basic_lti_launch(self):
         ).document_url
         return self.basic_lti_launch(document_url)
 
+    @view_config(db_configured=True, request_param="ext_lti_assignment_id")
+    def canvas_db_configured_basic_lti_launch(self):
+        """Respond to a Canvas DB-configured assignment launch."""
+        tool_consumer_instance_guid = self.request.params["tool_consumer_instance_guid"]
+        resource_link_id = self.request.params["resource_link_id"]
+        ext_lti_assignment_id = self.request.params["ext_lti_assignment_id"]
+
+        assignments = self.assignment_service.get_for_canvas_launch(
+            tool_consumer_instance_guid, resource_link_id, ext_lti_assignment_id
+        )
+
+        if len(assignments) == 2:
+            # We found two assignments: one with the matching resource_link_id and no ext_lti_assignment_id
+            # and one with the matching ext_lti_assignment_id and no resource_link_id.
+            #
+            # This happens because legacy code used to store Canvas assignments in the DB with a
+            # resource_link_id and no ext_lti_assignment_id, see https://github.com/hypothesis/lms/pull/2780
+            #
+            # Whereas current code stores Canvas assignments during content-item-selection with an
+            # ext_lti_assignment_id and no resource_link_id.
+            #
+            # We need to merge the two assignments into one.
+
+            # order is guaranteed by the query's `order by`
```

Comment:
```suggestion
```

I think we should remove the comment as it's unclear that _the query_ refers to (since it refers to a SQLAlchemy query in another file). It's part of `get_for_canvas_launch()`'s contract/docstring that if it returns two assignments it'll be the old one first: https://github.com/hypothesis/lms/blob/a6b7e551289181a9acb862b29aa9b4eef055f2ee/lms/services/assignment.py#L74